stl/inc/deque

```diff
@@ -772,35 +790,63 @@ public:
         emplace_front(_STD move(_Val));
     }
 
+#ifdef __cpp_lib_containers_ranges
+    template <_Container_compatible_range<_Ty> _Rng>
+    void prepend_range(_Rng&& _Range) {
+        _Orphan_all();
+
+        const auto _Oldsize = _Mysize();
+        _Restore_old_size_guard<_Pop_direction::_Front> _Guard{this, _Oldsize};
+        if constexpr (_RANGES bidirectional_range<_Rng>) {
+            auto _ULast        = _RANGES _Get_final_iterator_unwrapped(_Range);
+            const auto _UFirst = _RANGES _Ubegin(_Range);
+            while (_UFirst != _ULast) {
+                _Emplace_front_internal(*--_ULast); // prepend in order
+            }
+        } else {
+            auto _UFirst      = _RANGES _Ubegin(_Range);
+            const auto _ULast = _RANGES _Uend(_Range);
+            for (; _UFirst != _ULast; ++_UFirst) {
+                _Emplace_front_internal(*_UFirst); // prepend flipped
+            }
+
+            const auto _Num = static_cast<difference_type>(_Mysize() - _Oldsize);
+            _STD reverse(begin(), begin() + _Num); // flip new stuff in place

```

Comment:
I observe that this reversal is using checked `begin()`, but since we're forming this range ourselves, we could easily use `_Unchecked_begin()`. `deque` appears to be currently inconsistent about using unchecked iterators internally whenever possible - I see that some callsites do, but not all. Consider this to be "changes optionally requested"; I wouldn't object if you want to keep this as-is for now.

Reply:
Yes, I noticed the inconsistency. I think I'll leave this as is and file an issue to audit the entire file to consistently use unchecked iterators (#2868).